plugins/outputs/azure_monitor/azure_monitor.go

```diff
@@ -0,0 +1,491 @@
+	n := binary.PutUvarint(b, uint64(m.Time().UnixNano()))
+	h.Write(b[:n])
+	h.Write([]byte("\n"))
+	return h.Sum64()
+}
+
+func translate(m telegraf.Metric, prefix string) *azureMonitorMetric {
+	var dimensionNames []string
+	var dimensionValues []string
+	for i, tag := range m.TagList() {
+		// Azure custom metrics service supports up to 10 dimensions
+		if i > 10 {
+			log.Printf("W! [outputs.azure_monitor] metric [%s] exceeds 10 dimensions", m.Name())
+			continue
+		}
+		dimensionNames = append(dimensionNames, tag.Key)
+		dimensionValues = append(dimensionValues, tag.Value)
+	}
+
+	min, _ := m.GetField("min")
+	max, _ := m.GetField("max")
+	sum, _ := m.GetField("sum")
+	count, _ := m.GetField("count")
+	return &azureMonitorMetric{
+		Time: m.Time(),
+		Data: &azureMonitorData{
+			BaseData: &azureMonitorBaseData{
+				Metric:         strings.SplitN(m.Name(), "-", 2)[1],
+				Namespace:      prefix + strings.SplitN(m.Name(), "-", 2)[0],
+				DimensionNames: dimensionNames,
+				Series: []*azureMonitorSeries{
+					&azureMonitorSeries{
+						DimensionValues: dimensionValues,
+						Min:             min.(float64),
+						Max:             max.(float64),
+						Sum:             sum.(float64),
+						Count:           count.(int64),
+					},
+				},
+			},
+		},
+	}
+}
+
+// Add will append a metric to the output aggregate
+func (a *AzureMonitor) Add(m telegraf.Metric) {
+	// Azure Monitor only supports aggregates 30 minutes into the past and 4
+	// minutes into the future. Future metrics are dropped when pushed.
+	t := m.Time()
+	tbucket := time.Date(t.Year(), t.Month(), t.Day(), t.Hour(), t.Minute(), 0, 0, t.Location())
+	if tbucket.Before(time.Now().Add(-time.Minute * 30)) {
+		log.Printf("W! attempted to aggregate metric over 30 minutes old: %v, %v", t, tbucket)
+		return
+	}
+
+	// Azure Monitor doesn't have a string value type, so convert string fields
+	// to dimensions (a.k.a. tags) if enabled.
+	if a.StringsAsDimensions {
+		for fk, fv := range m.Fields() {
+			if v, ok := fv.(string); ok {
+				m.AddTag(fk, v)
+			}
+		}
+	}
+
+	for _, f := range m.FieldList() {
+		fv, ok := convert(f.Value)
+		if !ok {
+			continue
+		}
+
+		// Azure Monitor does not support fields so the field name is appended
+		// to the metric name.
+		name := m.Name() + "-" + sanitize(f.Key)
+		id := hashIDWithField(m.HashID(), f.Key)
+
+		_, ok = a.cache[tbucket]
+		if !ok {
+			// Time bucket does not exist and needs to be created.
+			a.cache[tbucket] = make(map[uint64]*aggregate)
+		}
+
+		nf := make(map[string]interface{}, 4)
+		nf["min"] = fv
+		nf["max"] = fv
+		nf["sum"] = fv
+		nf["count"] = 1
+		// Fetch existing aggregate
+		agg, ok := a.cache[tbucket][id]
+		if ok {
+			aggfields := agg.Fields()
+			if fv > aggfields["min"].(float64) {
+				nf["min"] = aggfields["min"]
+			}
+			if fv < aggfields["max"].(float64) {
+				nf["max"] = aggfields["max"]
+			}
+			nf["sum"] = fv + aggfields["sum"].(float64)
+			nf["count"] = aggfields["count"].(int64) + 1
```

Comment:
I haven't had a chance to look into this or test on i386 aside. It does not panic on x86_64 builds but I haven't had a chance to test on i386.

Reply:
You won't need to run this on a 32-bit system to test it, just compile with GOARCH=386 and try running it.  Looking at it again I guess it will work because we always convert int to int64 when adding them to a metric, but I do feel like this is brittle and a slight change to metric will introduce a panic here.